File: Java/lab19_Observer/task_2_1/src/WeatherEvent.java
public class WeatherEvent {

    final private String location;
    private final float temperature; // температура повітря
    private final float humidity;    // вологість повітря
    private final float pressure;    // атмосферний тиск

    public WeatherEvent(String location, float temperature, float humidity, float pressure) {
        this.location = location;
        this.temperature = temperature;
        this.humidity = humidity;
        this.pressure = pressure;
    }

    public String getLocation() {
        return location;
    }

    public float getTemperature() {
        return temperature;
    }

    public float getHumidity() {
        return humidity;
    }

    public float getPressure() {
        return pressure;
    }
}

File: Cpp/lab19_Observer/task_2_1 - solution/src/weather_event.cpp
#include "weather_event.hpp"


WeatherEvent::WeatherEvent(std::string const& location, float temperature, float humidity, float pressure)
: location(location), temperature(temperature), humidity(humidity), pressure(pressure) {}

std::string const& WeatherEvent::getLocation() const
{
  return this->location;
}

float WeatherEvent::getTemperature() const
{
  return this->temperature;
}

float WeatherEvent::getHumidity() const
{
  return this->humidity;
}

float WeatherEvent::getPressure() const
{
  return this->pressure;
}
